modules/node_groups/variables.tf

```diff
@@ -0,0 +1,58 @@
+variable "create_eks" {
+  description = "Controls if EKS resources should be created (it affects almost all resources)"
+  type        = bool
+  default     = true
+}
+
+variable "cluster_name" {
+  description = "Name of parent cluster"
+  type        = string
+}
+
+variable "cluster_version" {
+  description = "Kubernetes version of parent cluster"
+  type        = string
+}
+
+variable "default_iam_role_arn" {
+  description = "ARN of the default IAM worker role to use if one is not specified in the node_groups"
+  type        = string
+}
+
+variable "workers_group_defaults" {
+  description = "Workers group defaults from parent"
+  type        = any
+}
+
+variable "tags" {
+  description = "A map of tags to add to all resources"
+  type        = map(string)
+}
+
+variable "node_groups_defaults" {
+  description = "map of maps of node groups to create. See default for valid keys and type. See source for extra comments"
+  type        = any
+
+  # This will be always be overriden by the value from the top level module
+  default = {
+    iam_role_arn              = ""           # IAM Role ARN for workers. If unset: uses `var.default_iam_role_arn`
```

Comment:
By defining defaults here, can we just overwrite one (or some) default(s) ?

Reply:
I now fear that this attempt at abusing automatic documentation may actually cause more confusion than it solves 😅 

The values in this default map are rubbish. And will cause errors/issues if actually used. They're to indicate the set of expected keys and their data types. When only used by the parent module this will always be set to an empty map or whatever the user passes in. This default will need removing if we're expecting users to directly use the module. 

Maybe the whole lot should be moved to README.md. Then the extended comments will be more visible too without having to dive into the source code.

Reply:
Made changes and moved this weird default and the commends into the module's README.